src/agentic/actor_agents.py

```diff
@@ -349,44 +362,22 @@ async def run_async_gen():
 
         return partial_response, events
 
-    def _init_request_state(self, request_id: str) -> dict[str, Any]:
-        """Initialize state for a new request"""
-        state = {
-            'paused_context': None,
-            'debug': DebugLevel(False),
-            'depth': 0,
-            'run_context': None,
-            'init_history_len': len(self.history)  # Track where this request started
-        }
-        self.request_states[request_id] = state
-        return state
-
-    def _get_request_state(self, request_id: str) -> dict[str, Any]:
-        """Get state for a request, initializing if needed"""
-        return self.request_states.get(request_id) or self._init_request_state(request_id)
-
     def handlePromptOrResume(self, actor_message: Prompt | ResumeWithInput):
         request_id = getattr(actor_message, 'request_id', None)
         if not request_id:
+            print("Missing request id from actor_message: ", actor_message, " with type: ", type(actor_message))
             raise ValueError("Request ID is required")
                     
-        try:
-            for event in self._handlePromptOrResume(actor_message, request_id):
-                if self._callbacks.get('handle_event'):
-                    state = self._get_request_state(request_id)
-                    self._callbacks['handle_event'](event, state['run_context'])
-                yield event
-        finally:
-            # Only clean up transient state
-            if request_id in self.request_states:
-                del self.request_states[request_id]
```

Comment:
Need to review the logic for destroying an agent instance inside the proxy once the request is done. Do we need that?